### PR TITLE
Made it possible for the TO_* functions to take more varied arguments

### PR DESCRIPTION
**Pull request by magnusnordlander**
We want to be able to pass any kind of StringPrimary into the `TO_JSONB` function, however, I think it makes sense to be able to pass e.g. a SimpleArithmeticExpression as well, and it seems NewValue is the best corresponding type for that.

Also, even though we're only using it in `TO_JSONB`, obviously this makes sense for the other `TO_*` functions too.

### CODE REVIEW

**Comment by martin-georgiev** [comment]
Would you please new [tests](https://github.com/martin-georgiev/postgresql-for-doctrine/blob/develop/tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/ToTsqueryTest.php) for this?

**Comment by magnusnordlander** [comment]
Sure, I have now added test cases for some of the other parser types that NewValue allows.

File: tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/ToTsvectorTest.php [hunk 1]
```diff
@@ -20,13 +20,21 @@ protected function getExpectedSqlStatements(): array
     {
         return [
             'SELECT to_tsvector(c0_.text) AS sclr_0 FROM ContainsText c0_',
+            'SELECT to_tsvector(LOWER(c0_.text)) AS sclr_0 FROM ContainsText c0_',
+            'SELECT to_tsvector(1 + 1) AS sclr_0 FROM ContainsText c0_',
+            'SELECT to_tsvector(1) AS sclr_0 FROM ContainsText c0_',
+            'SELECT to_tsvector(LENGTH(c0_.text)) AS sclr_0 FROM ContainsText c0_',
```
Comment (martin-georgiev) [inline tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/ToTsvectorTest.php]: The checks from lines 24, 25 and 26 won't produce meaningful SQL, as _to_tsvector(integer)_ is not a valid function. You can safely drop those scenarios.

File: tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/ToTsqueryTest.php [hunk 2]
```diff
@@ -20,13 +20,21 @@ protected function getExpectedSqlStatements(): array
...
     }
 
     protected function getDqlStatements(): array
     {
         return [
             \sprintf('SELECT TO_TSQUERY(e.text) FROM %s e', ContainsText::class),
+            \sprintf('SELECT TO_TSQUERY(UPPER(e.text)) FROM %s e', ContainsText::class),
+            \sprintf('SELECT TO_TSQUERY(1+1) FROM %s e', ContainsText::class),
+            \sprintf('SELECT TO_TSQUERY(true) FROM %s e', ContainsText::class),
+            \sprintf('SELECT TO_TSQUERY(LENGTH(e.text)) FROM %s e', ContainsText::class),
```
Comment (martin-georgiev) [inline tests/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/ToTsqueryTest.php]: The checks from lines 35, 36 and 37 won't produce meaningful SQL, as to_tsquery(integer) is not a valid function. You can safely drop those scenarios.

**Comment by magnusnordlander** [comment]
I suppose that considering that to_tsquery and to_tsvector only takes strings, this doesn't make much sense for them, so i've undone those changes. I added the same tests for to_json and to_jsonb though, since the accept anyelement.

**Comment by martin-georgiev** [comment]
Thank you @magnusnordlander for this contribution! It's lovely to see new people enhancing the project.

**Pull request merged**